### ISSUE TITLE
Brave failed to install after uninstalling if Wireguard protocol is enabled and VPN is connected 

### ISSUE THREAD

**Issue by MadhaviSeelam** [brave/brave-browser#33182]
## Description 

Found the issue while testing https://github.com/brave/brave-browser/issues/32523.  It looks like brave_vpn_wireguard_service.exe interferes with the installation.  Once enabling Wireguard protocol and connecting to VPN, closing, uninstalling and re-installing fails Brave unless brave_vpn_wireguard_service is killled.

## Steps to Reproduce

1. Install 1.59.94
2. launch Brave
3. open brave://settings/system
4. `Use Wireguard protocol in Brave VPN` is enabled by default
5. purchased, configured and connected to Brave VPN
6. disconnected VPN in the VPN panel
7. opened cmd line prompt as an administrator and ran sc.exe delete BraveBetaVpnWireguardService
8. connected to VPN via Brave VPN panel
9. install 1.59.94

## Actual result:

Installation failed due to unspectified error...` message shown

https://github.com/brave/brave-browser/assets/98358127/ff465f36-5f3d-4edf-a3f1-31d3365bddd1

## Expected result:
Installation should succeed.

## Reproduces how often: 

Easily (when Wireguard protocol enabled and VPN is connected)

## Brave version (brave://version info)

```
Brave | 1.59.94 Chromium: 117.0.5938.88 (Official Build) beta (64-bit)
-- | --
Revision | 2dca22d0a20096c32df5af4c88700a0b31acb55c
OS | Windows 11 Version 22H2 (Build 22621.2283)
```
## Version/Channel Information:

- Can you reproduce this issue with the current release? N/A
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? N/A

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:
<!--Any additional information, related issues, extra QA steps, configuration or data that might be necessary to reproduce the issue--
@bsclifton @spylogsster 
cc: @stephendonner

**Comment by GeetaSarvadnya** [comment]
Verification **INPROGRESS** on

```
Brave | 1.60.81 Chromium: 118.0.5993.54 (Official Build) beta (64-bit)
-- | --
Revision | 1efb3f333eb41cc34af46ca31fb33c30e0afbfae
OS | Windows 10 Version 22H2 (Build 19045.3448)
```

**Comment by stephendonner** [comment]
I also can reproduce the above using 

```
Brave | 1.60.93 Chromium: 118.0.5993.70 (Official Build) beta (64-bit)
-- | --
Revision | 71222cfbe432cb4f3275086ad00abe4eb428eafe
OS | Windows 10 Version 22H2 (Build 19045.3570)
```

I believe it might be the `Brave VPN` system-tray notification icon, which was still running, even after I ran `sc.exe delete BraveBetaVpnWireguardService`, but of course I'm not sure.

**Comment by bsclifton** [comment]
@stephendonner great eye there 😄 That is exactly the problem

I went to verify this and found an issue. The way the issue is worded, everything works fine. After you connect to VPN after deleting the service using `sc.exe delete BraveBetaVpnWireguardService`, it falls back to IKEv2 as it should:
![image](https://github.com/brave/brave-browser/assets/4733304/b29e0fb3-c125-4c30-97b0-9fbc50110dec)

If the complaint is that uninstalling after switching from WireGuard over to IKEv2, problem is solved and fix is verified 😄👍 

As the steps are captured though, a different situation is happening and that issue is NOT resolved. I opened https://github.com/brave/brave-browser/issues/33904 to capture this one in better detail.

The tray program was also updated to support IKEv2 (see https://github.com/brave/brave-browser/issues/32143). When we (as QA) are going and deleting the WireGuard service via `sc.exe delete BraveBetaVpnWireguardService`, **this means the cleanup for the tray icon doesn't happen**. Even though it's IKEv2, the tray program itself is still using the `BraveBetaWireguardService.exe` binary.

Hope this makes sense! 😄 It's an edge case for sure
